### PR TITLE
Add aws-cloud-controller-manager [1/2]

### PR DESCRIPTION
**Pull request by mikkeloscar**
Add aws-cloud-controller-manager daemonset to run on master nodes.

### CODE REVIEW

File: cluster/manifests/aws-cloud-controller-manager/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cloud-controller-manager
+  namespace: kube-system
+  labels:
+    application: kubernetes
+    component: aws-cloud-controller-manager
```
Comment (linki) [inline cluster/manifests/aws-cloud-controller-manager/rbac.yaml:9]: Don't we need an IAM role for it or how does it get permission to access AWS?

Comment (mikkeloscar) [reply cluster/manifests/aws-cloud-controller-manager/rbac.yaml:9]: Good point!

It uses `hostNetwork` (taken from upstream example) so it's using the role of the node, but it makes sense to use a dedicated role ofc.

Comment (mikkeloscar) [reply cluster/manifests/aws-cloud-controller-manager/rbac.yaml:9]: Changed to use dedicated IAM role and not `hostNetwork`.

**Comment by mikkeloscar** [comment]
:+1:

File: cluster/config-defaults.yaml [hunk 2]
```diff
@@ -428,6 +428,12 @@ kubernetes_lifecycle_metrics_mem_min: "120Mi"
 kube_node_ready_controller_cpu: "50m"
 kube_node_ready_controller_memory: "200Mi"
 
+# Enable deployment of aws-cloud-controller-manager
+aws_cloud_controller_manager_enabled: "true"
```
Comment (demonCoder95) [inline cluster/config-defaults.yaml:432]: Shouldn't we keep it `false` by default as we're testing it? 🤔  Or do we want to control the testing/rollout by channel and not by cluster?

Comment (mikkeloscar) [reply cluster/config-defaults.yaml:432]: We can flip the logic to false and enable per cluster. This is coming from the perspective, it does seem to work in e2e, so I assume it works and we can disable it if it does not.

But changing to false and enabling manually is also fine by me.

**Comment by linki** [comment]
👍

File: cluster/config-defaults.yaml (hunk 2 above)
Comment (demonCoder95) [reply cluster/config-defaults.yaml:432]: Makes sense to me then. Thanks for explaining! :)

**Comment by demonCoder95** [comment]
Thanks for merging! 🙏

**Pull request merged**